### PR TITLE
UPSTREAM: Allow getting logs and attach for multiple resources

### LINKED ISSUE #12789: Deployments don't work with oc logs
```
$ oc logs deployment/prometheus
error: cannot get the logs from extensions/__internal, Kind=Deployment
```

Should work.

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #12789.

@smarterclayton waiting for upstream approval

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go [hunk 1]
```diff
@@ -222,6 +222,26 @@ func (f *ring1Factory) LogsForObject(object, options runtime.Object) (*restclien
 
 		return clientset.Core().Pods(pod.Namespace).GetLogs(pod.Name, opts), nil
 
+	case *extensions.Deployment:
```
Comment (smarterclayton) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go]: Also stateful sets!  and jobs!  and cron jobs!

Comment (soltysh) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go]: OK for stateful sets and jobs, not sure about cron jobs. Lemme see...

Comment (soltysh) [inline vendor/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory_object_mapping.go]: Nope for cron jobs, at least for now.

**Comment by soltysh** [comment]
@smarterclayton I've updated this PR with the current state of the upstream PR, additionally I've cherry-picked the other PR which fixes analogical problem, but for attach. ptal

[test]

**Comment by soltysh** [comment]
This is not going to happen in 1.5 and master will get this with the next rebase, so I'm closing this.

**Pull request closed without merging**